### PR TITLE
Add test to check if swap is enabled

### PR DESCRIPTION
**Pull request by stefan-caraiman**
Adds a new test scenario for checking whether the swap memory is enabled
on the underlying instance.

### CODE REVIEW

File: argus/introspection/cloud/windows.py [hunk 1]
```diff
@@ -397,3 +397,17 @@ def get_user_flags(self, user):
...
+        """Get whether the swap memory is enabled or not.
+
+        :returns: True if swap memory is enabled, False if not.
+        :rtype: bool
+        """
+        swap_query = (r"'HKLM:\SYSTEM\CurrentControlSet\Control\Session"
+                      r" Manager\Memory Management'")
+        cmd = r'(Get-ItemProperty "{}").ExistingPageFiles'.format(swap_query)
+        stdout = self.remote_client.run_command_verbose(cmd)
+        if not stdout:
```
Comment (ader1990) [inline argus/introspection/cloud/windows.py]: add a check if stdout equals to ?:\pagefile.sys

File: ci/tests.py [hunk 2]
```diff
@@ -231,3 +231,8 @@ class ScenarioImageSmoke(ScenarioSmoke):
 
     recipe_type = recipe.CloudbaseinitImageRecipe
     userdata = util.get_resource("windows/winrm.ps1")
+
```
Comment (ader1990) [inline ci/tests.py]: don t create a new scenario, add the test to secnarioimagesmoke

**Review by ader1990 — CHANGES_REQUESTED** [review]

File: argus/introspection/cloud/windows.py [hunk 3]
```diff
@@ -397,3 +397,17 @@ def get_user_flags(self, user):
...
+
+    def get_swap_status(self):
+        """Get whether the swap memory is enabled or not.
+
+        :returns: True if swap memory is enabled, False if not.
+        :rtype: bool
+        """
+        swap_query = (r"'HKLM:\SYSTEM\CurrentControlSet\Control\Session"
+                      r" Manager\Memory Management'")
+        cmd = r'(Get-ItemProperty {}).PagingFiles'.format(swap_query)
```
Comment (mateimicu) [inline argus/introspection/cloud/windows.py]: This is just a cosmetic thing, but you could have the `swap_query` without single quotes and add them in the `cmd` string . `(Get-ItemProperty '{}')...`

File: argus/introspection/cloud/windows.py [hunk 4]
```diff
@@ -397,3 +397,17 @@ def get_user_flags(self, user):
...
+        stdout = self.remote_client.run_command_verbose(cmd)
+        if stdout == '?:\pagefile.sys':
```
Comment (mateimicu) [inline argus/introspection/cloud/windows.py]: Another cosmetic thing is to rewrite this two lines like this
 `return stdout == '?:\pagefile.sys'`.
And you may want to strip the `stdout` something like 
`return stdout.strip() == '?:\pagefile.sys'`

**Review by mateimicu — CHANGES_REQUESTED** [review]

File: argus/introspection/cloud/windows.py (hunk 4 above)
Comment (stefan-caraiman) [inline argus/introspection/cloud/windows.py]: The stdout can be `None` too if the key has not been found, so the strip approach will fail in that case, unless we first check if it's `None` or not.

**Review by ader1990 — APPROVED** [review]
lgtm

File: argus/tests/cloud/smoke.py [hunk 5]
```diff
@@ -251,6 +251,15 @@ def test_hostname_compatibility(self):
         self.assertEqual("someverylonghostnametosetonthemachine-00", hostname)
 
 
+class TestSwapEnabled(base.BaseTestCase):
+
+    def test_swap_enabled(self):
```
Comment (alexcoman) [inline argus/tests/cloud/smoke.py:257]: Add docstring for this test.

File: argus/tests/cloud/smoke.py [hunk 6]
```diff
@@ -251,6 +251,15 @@ def test_hostname_compatibility(self):
         self.assertEqual("someverylonghostnametosetonthemachine-00", hostname)
 
 
+class TestSwapEnabled(base.BaseTestCase):
```
Comment (alexcoman) [inline argus/tests/cloud/smoke.py:254]: Add docstring.

**Review by alexcoman — CHANGES_REQUESTED** [review]

**Review by alexcoman — APPROVED** [review]

**Pull request merged**